integrations/sonarqube/main.py

```diff
@@ -18,14 +25,73 @@ def init_sonar_client() -> SonarQubeClient:
     )
 
 
+def produce_project_params(
+    api_filter: SonarQubeProjectApiFilter | None,
+) -> dict[str, Any]:
+    project_params: dict[str, Any] = {}
+    if not api_filter:
+        return project_params
+
+    if api_filter.analyzed_before:
+        project_params["analyzedBefore"] = api_filter.analyzed_before
+
+    if api_filter.on_provisioned_only:
+        project_params["onProvisionedOnly"] = api_filter.on_provisioned_only
+
+    if api_filter.projects:
+        project_params["projects"] = ",".join(api_filter.projects)
+
+    if api_filter.qualifiers:
+        project_params["qualifiers"] = ",".join(api_filter.qualifiers)
+
```

Comment:
the SonarQubeProjectApiFilter  has a `generate_request_params()`, is there any reason why you're not reusing that method since you've even made the same logic there?

```
        if self.projects:
            value["projects"] = ",".join(self.projects)

        if self.qualifiers:
            value["qualifiers"] = ",".join(self.qualifiers)
```

Reply:
I realized that I'd have to explicitly pop out the other items I didn't need. Since I have already included the hard coded strings "projects", "qualifiers" and so on  in the next function below, explicitly stating the other properties here to be removed would result in me specifying all the properties in the `integration` file here. So I decided to only have these properties hard coded here .